nightfall-client/src/services/commitment-storage.mjs

```diff
@@ -688,3 +688,38 @@ export async function findUsableCommitmentsMutex(
     findUsableCommitments(compressedZkpPublicKey, ercAddress, tokenId, _value, onlyOne),
   );
 }
+
+/**
+ * @function getCommitmentsByCompressedZkpPublicKeyList do the role of a service taking care of the
+ * business logic and of a repository doing the communication with the database for this
+ * use case.
+ * @param {string[]} listOfCompressedZkpPublicKey a list of compressedZkpPublicKey derivated from the user
+ * mnemonic coming from the SDK or Wallet.
+ * @returns all the commitments existent for this list of compressedZkpPublicKey.
+ * @author luizoamorim
+ */
+export async function getCommitmentsByCompressedZkpPublicKeyList(listOfCompressedZkpPublicKey) {
+  const connection = await mongo.connection(MONGO_URL);
+  const db = connection.db(COMMITMENTS_DB);
+  const commitmentsByListOfCompressedZkpPublicKey = await db
+    .collection(COMMITMENTS_COLLECTION)
+    .find({
+      'preimage.compressedZkpPublicKey': { $in: listOfCompressedZkpPublicKey },
+    })
+    .toArray();
+  return commitmentsByListOfCompressedZkpPublicKey;
```

Comment:
I will insist one more time about returning directly from L704 :upside_down_face: (in case you missed the comment!?)

Reply:
Eslint complains about returning from that line.